### PR TITLE
make KV immutable and always valid

### PR DESCRIPTION
**Pull request by kopiczko**
This also required introduction of new type K, for querying functions.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -2,25 +2,113 @@ package microstorage
...
+	RootKey = K{key: "/"}
 )
 
-// KV is a key-value pair.
+// K is an immuatable, valid key.
+type K struct {
+	key string
+}
+
+// NewK creates a new immutable key to quiery the Storage with.
```
Comment (JosephSalisbury) [inline storage.go]: `query`

File: storage.go [hunk 2]
```diff
@@ -2,25 +2,113 @@ package microstorage
...
+	key string
+	val string
+}
+
+// NewKV creates a new immutable key-value pair to update Storage with.
+//
+// The key should take a format of path separated by slashes "/". E.g.
+// "/a/b/c".  The key is sanitized before inserting to the storage. I.e.
+// leading slash can be added and trailing slash can be removed.  E.g.
+// "/a/b/c", "a/b/c/", "a/b/c/", and "/a/b/c/" represent the same key.
```
Comment (JosephSalisbury) [inline storage.go]: `a/b/c/` is here twice?

Comment (kopiczko) [inline storage.go]: One should be `a/b/c`

**Comment by kopiczko** [comment]
@JosephSalisbury happy :) 🎉 
Now I'd like to know what @xh3b4sd has to say.

**Comment by kopiczko** [comment]
Cool, going with it.

**Pull request merged**